### PR TITLE
LPA-3483 - Create dns records for any stack in the claim hosted zone

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
- add credentials for old refunds account
- create user facing dns records for the claim and caseworker on any/every environment for parity
- update load balancer rules to redirect only the claim dns record to maintenance when switched
- generate and attach ssl certificates without the account namspacing
- publish user facing urls in slack messages
- add branch name and last commit message to slack messages
- On production, user facing dns still points too old environment cloudfront and caseworker services

### CODE REVIEW

File: terraform/account/terraform.tfvars.json [hunk 1]
```diff
@@ -9,20 +9,20 @@
       "account_id": "936779158973",
       "is_production": false,
       "has_cloudfront_distribution": false,
-      "public_front_certificate_domain_name": "*.public-front.development.refunds.opg.service.justice.gov.uk",
-      "caseworker_front_certificate_domain_name": "*.caseworker.development.refunds.opg.service.justice.gov.uk",
-      "public_front_dns": "public-front.development.refunds",
-      "caseworker_front_dns": "caseworker.development.refunds",
+      "public_front_certificate_domain_name": "*.public-front.refunds.opg.service.justice.gov.uk",
```
Comment (damacus) [inline terraform/account/terraform.tfvars.json]: public-front --> makes me so sad :(

File: terraform/environment/dns_claim.tf [hunk 2]
```diff
@@ -0,0 +1,62 @@
+data "aws_route53_zone" "claim-power-of-attorney-refund_service_gov_uk" {
+  provider = aws.old_refunds_production
+  name     = "claim-power-of-attorney-refund.service.gov.uk"
+}
+
+locals {
+  public_front_cloudfront_distribution_domain_name = "d3pm2jfxjwrqjc.cloudfront.net"
+  public_front_cloudfront_distribution_zone_id     = "Z2FDTNDATAQYW2"
```
Comment (damacus) [inline terraform/environment/dns_claim.tf]: Could this not be a data source?

File: terraform/environment/dns_claim.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+  }
+  claim_url_dns_target = local.account == "production" ? local.claim_url_dns_target_production : local.claim_url_dns_target_dev_preprod
+}
+
+
+
+//-------------------------------------------------------------
+// Claim - points to public front
+resource "aws_route53_record" "claim-power-of-attorney-refund_service_gov_uk" {
```
Comment (damacus) [inline terraform/environment/dns_claim.tf]: Mixing `_` with `-` feels/looks-wrong

File: terraform/environment/.envrc [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export TF_VAR_default_role=operator
+export TF_VAR_old_account_default_role=operator
 export TF_CLI_ARGS_init="-backend-config=role_arn=arn:aws:iam::311462405659:role/operator -upgrade=true  -reconfigure"
-export TF_WORKSPACE=
+export TF_WORKSPACE=
```
Comment (hansnqyr) [inline terraform/environment/.envrc]:
```suggestion
export TF_WORKSPACE=

```

File: terraform/environment/locals.tf [hunk 5]
```diff
@@ -31,7 +31,7 @@ locals {
   account_name      = lookup(var.account_mapping, terraform.workspace, "development")
   account           = var.accounts[local.account_name]
   environment       = lower(terraform.workspace)
-  dns_namespace_env = local.account_name != "development" ? "" : "${local.environment}."
+  dns_namespace_env = local.account_name == "production" ? "" : "${local.environment}."
```
Comment (hansnqyr) [inline terraform/environment/locals.tf]: 😢

File: .circleci/config.yml [hunk 6]
```diff
@@ -570,9 +570,14 @@ jobs:
           command: |
             echo 'export FRONT_DOMAIN="$(jq -r .public_front_fqdn /tmp/environment_pipeline_tasks_config.json)"' >> $BASH_ENV
             echo 'export CASEWORKER_DOMAIN="$(jq -r .caseworker_front_fqdn /tmp/environment_pipeline_tasks_config.json)"' >> $BASH_ENV
+            echo 'export CLAIM_SERVICE_GOV_UK="$(jq -r .claim_service_gov_uk_fqdn /tmp/environment_pipeline_tasks_config.json)"' >> $BASH_ENV
+            echo 'export CASEWORKER_REFUNDS_OPG_DIGITAL="$(jq -r .caseworker_refunds_opg_digital_fqdn /tmp/environment_pipeline_tasks_config.json)"' >> $BASH_ENV
+            echo 'export COMMIT_MESSAGE="$(git log -1 --pretty=%B)"' >> $BASH_ENV
```
Comment (hansnqyr) [inline .circleci/config.yml]: I wonder if a small script here could tidy this up a bit?

File: terraform/environment/locals.tf (hunk 5 above)
Comment (andrewpearce-digital) [reply terraform/environment/locals.tf]: This conditional will disappear after ecs migration

File: terraform/environment/dns_claim.tf (hunk 2 above)
Comment (andrewpearce-digital) [reply terraform/environment/dns_claim.tf]: No data source exists for cloudfront distrbutions

File: terraform/environment/dns_caseworker.tf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  caseworker_url_dns_target_production = {
+    alias = {
+      name    = data.aws_elb.old_production_caseworker_front.dns_name
+      zone_id = data.aws_elb.old_production_caseworker_front.zone_id
+    }
+  }
+  #this condition is failing to select the correct set of load balancers
+  caseworker_url_dns_target = local.account_name == "production" ? local.caseworker_url_dns_target_production : local.caseworker_url_dns_target_dev_preprod
```
Comment (hansnqyr) [inline terraform/environment/dns_caseworker.tf]: is this switching between old and new production formats?

Comment (andrewpearce-digital) [reply terraform/environment/dns_caseworker.tf]: it's switching between old and new production targets

File: terraform/account/terraform.tfvars.json [hunk 8]
```diff
@@ -9,20 +9,20 @@
...
       "is_production": false,
       "has_cloudfront_distribution": true,
-      "public_front_certificate_domain_name": "public-front.preproduction.refunds.opg.service.justice.gov.uk",
-      "caseworker_front_certificate_domain_name": "caseworker.preproduction.refunds.opg.service.justice.gov.uk",
-      "public_front_dns": "public-front.preproduction.refunds",
-      "caseworker_front_dns": "caseworker.preproduction.refunds",
+      "public_front_certificate_domain_name": "*.public-front.refunds.opg.service.justice.gov.uk",
+      "caseworker_front_certificate_domain_name": "*.caseworker.refunds.opg.service.justice.gov.uk",
+      "public_front_dns": "public-front.refunds",
+      "caseworker_front_dns": "caseworker.refunds",
```
[the remark this answers is not in the document]
Comment (andrewpearce-digital) [reply terraform/account/terraform.tfvars.json]: this has been reduced from 4 to 0

**Pull request merged**